lib/stripe_mock/request_handlers/payment_intents.rb

```diff
@@ -18,17 +22,19 @@ def new_payment_intent(route, method_url, params, headers)
 
         ensure_payment_intent_required_params(params)
         status = case params[:amount]
-        when 3184 then 'requires_action'
-        when 3178 then 'requires_payment_method'
+        when REQUIRES_ACTION_AMOUNT then 'requires_action'
+        when FAILED_TRANSACTION_AMOUNT then 'requires_payment_method'
+        when REQUIRES_CAPTURE_AMOUNT then 'requires_capture'
         else
           'succeeded'
         end
-        last_payment_error = params[:amount] == 3178 ? last_payment_error_generator(code: 'card_declined', decline_code: 'insufficient_funds', message: 'Not enough funds.') : nil
+        last_payment_error = params[:amount] == FAILED_TRANSACTION_AMOUNT ? last_payment_error_generator(code: 'card_declined', decline_code: 'insufficient_funds', message: 'Not enough funds.') : nil
         payment_intents[id] = Data.mock_payment_intent(
           params.merge(
             id: id,
             status: status,
-            last_payment_error: last_payment_error
+            last_payment_error: last_payment_error,
+            charges: status == 'succeeded' ? charges_date : empty_charges
```

Comment:
You should also add charges in `confirm_payment_intent`

Reply:
good point!  Thanks! I'll update.

Reply:
Nice, thanks for the quick update !